platform/darwin/src/NSData+MGLAdditions.mm

```diff
@@ -0,0 +1,33 @@
+#import "NSData+MGLAdditions.h"
+
+#include <mbgl/util/compression.hpp>
+
+@implementation NSData (MGLAdditions)
+
+- (NSData *)mgl_compressedData
+{
+    NSString *dataString = [[NSString alloc] initWithData:self encoding:NSUTF8StringEncoding];
+    
+    std::string *string = new std::string([dataString UTF8String]);
```

Comment:
C++ types should be allocated on the stack, whenever possible, rather than the heap.